docs/run-command.md

```diff
@@ -43,42 +72,68 @@ default via 10.240.0.1 dev eth0 proto dhcp src 10.240.0.4 metric 100
-$ kubectl aks run-command "hostname"
+If we run another command, it will again be executed in `aks-agentpool-12345678-vmss000000`:
+
+```bash
+kubectl aks run-command "hostname"
+```
+
+The output should be similar to this:
+
+<!--expected_similarity=0.8-->
+```
 aks-agentpool-12345678-vmss000000
 ```
 
+Unset the current node to avoid conflict with flags or environment variables:
+
+```bash 
+kubectl aks config unset-current-node
+```
+
 On the other side, if we already have the node (VMSS instance) information and we don't want/need to save it locally, we could pass it directly as following:
 
-```bash
-kubectl aks run-command "ip route" --id "/subscriptions/$SUBSCRIPTION/resourceGroups/$NODERESOURCEGROUP/providers/Microsoft.Compute/virtualMachineScaleSets/$VMSS/virtualmachines/$INSTANCEID"
+<!-- TODO: Test following when we have a simple way to get instance information -->
+```
+kubectl aks run-command "ip route" --id "/subscriptions/$mySubID/resourceGroups/$myNRG/providers/Microsoft.Compute/virtualMachineScaleSets/$myVMSS/virtualmachines/$myInsId"
 ```
 
-```bash
-kubectl aks run-command "ip route" --subscription $SUBSCRIPTION --node-resource-group $NODERESOURCEGROUP --vmss $VMSS --instance-id $INSTANCEID
+Or using the flags:
+
+```
+kubectl aks run-command "ip route" --subscription $mySubID --node-resource-group $myNRG --vmss $myVMSS --instance-id $myInsId
 ```
 
 ## Passing through Kubernetes
 
 If we are debugging a node while the Kubernetes control plane is up and running, we can simply pass the node name to the `run-command` and it will internally retrieve all the data it needs from the API server to execute the command in that node:
 
 ```bash
-kubectl aks run-command "ip route" --node aks-agentpool-12345678-vmss000000
+kubectl aks run-command "ip route" --node $myNode
 ```
 
 In addition, if we need to run multiple commands on a node, we can still use the [`config import`](./config.md#importing-configuration) command to import the information of all the nodes of our cluster, and this time we don't need to pass the cluster information as `run-command` will retrieve it from the API server:
 
 ```bash
-# Import the nodes information from the API server
 kubectl aks config import
+```
+
+Start using one of the nodes e.g `aks-agentpool-12345678-vmss000000` we call it `$myNode` here:
+
+```bash
+kubectl aks config use-node $myNode
+```
 
-# Start using one of the nodes
-kubectl aks use-node aks-agentpool-12345678-vmss000000
+Execute the run-command, and it will be automatically executed in `aks-agentpool-12345678-vmss000000`:
 
-# Execute the run-command, and it will be automatically executed in aks-agentpool-12345678-vmss000000
+```bash
 kubectl aks run-command "ip route"
+```
+
+If we run another command, it will again be executed in `aks-agentpool-12345678-vmss000000`:
 
-# Another command that will be still executed in aks-agentpool-12345678-vmss000000
+```bash
 kubectl aks run-command "hostname"
 ```
```

Comment:
I don't know, it doesn't make sense to repeat everything here again. Probably, we should change the structure of this file:

- The two mechanisms to retrieve the info (and we verify the output from both using ExecDocs).
- Show the example (only once and no matter how we retrieve the info as we already tested that both are fine).

It's fine to manage this comment in later.